src/components/PreviewSection/Filter/Filter.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import {
+  Expand,
+  FilterContent,
+  FilterHeader,
+  FilterList,
+  FilterWrapper,
+  RegionTitle,
+  StateItem,
+  Title,
+} from "./Filter.styles";
+import { IRegionData } from "@/utils/interfaces";
+
+const Filter = ({
+  data,
+  selectedRegion,
+  selectedState,
+  onChange,
+}: {
+  data: IRegionData[];
+  selectedRegion: string;
+  selectedState: string;
+  onChange: (region: string, state: string) => void;
+}) => {
+  const regions = Array.from(new Set(data.map((card) => card.region)));
+
+  const getRegionStates = (region: string) => {
+    const regionCard = data.find((card) => card.region === region);
+
+    return regionCard ? regionCard.states : [];
+  };
+
+  return (
+    <FilterWrapper>
+      <FilterHeader>
+        <Title>{selectedState || selectedRegion}</Title>
+        <Expand id="expand" size={10} />
+      </FilterHeader>
+
+      <FilterContent>
+        <FilterList>
+          {regions.map((region) => (
+            <li key={region}>
+              <RegionTitle onClick={() => onChange(region, "")}>
+                {region}
+              </RegionTitle>
+
+              <ul>
```

Comment:
I think we should remove this ul tag and let the states for this be at the same level as the region's

```suggestion
```

I'm suggesting this because we'll never have other regions because sudene only works on Nordeste